src/music_service.py

```diff
@@ -0,0 +1,269 @@
+            self.send_position()
+            self.send_status()
+        else:
+            self.send_metadata(title="<select a track>",
+                               artist="Player: " + self.active_player_name.rsplit('.', 1)[1],
+                               album="",
+                               total_length=0)
+
+            self.send_position(0)
+            self.send_status(0)
+
+    def active_player_properties_changed(self, iface, changed_prop, invalid):
+        if 'Metadata' in changed_prop:
+            metadata = changed_prop['Metadata']
+            if 'xesam:title' in metadata:
+                self.send_metadata(metadata)
+                self.send_position()
+                self.send_status()
+            else: #end of a playlist
+                self.send_metadata(title="<select a track>",
+                                   artist="Player: " + self.active_player_name.rsplit('.', 1)[1],
+                                   album="",
+                                   total_length=0)
+                self.send_position(0)
+                self.send_status(0)
+        elif 'PlaybackStatus' in changed_prop:
+            self.send_status(changed_prop['PlaybackStatus'])
+
+    def characteristic_value_updated(self, chrc, value):
+        if self.active_player:
+            if chrc == self.event_chrc:
+                if value == self.EVENT_NEXT:
+                    self.player_iface.Next()
+
+                elif value == self.EVENT_PREV:
+                    self.player_iface.Previous()
+
+                elif value == self.EVENT_PLAY:
+                    self.player_iface.Play()
+
+                elif value == self.EVENT_PAUSE:
+                    self.player_iface.Pause()
+
+                elif value == self.EVENT_VOLDOWN:
+                    curr_volume = self.get_player_propertie('Volume')
+                    if curr_volume:
+                        self.set_player_propertie('Volume', curr_volume - self.VOLUME_STEP)
+
+                elif value == self.EVENT_VOLUP:
+                    curr_volume = self.get_player_propertie('Volume')
+                    if curr_volume:
+                        self.set_player_propertie('Volume', curr_volume + self.VOLUME_STEP)
+
+                elif value == self.EVENT_OPEN:
+                    #the current position is lost when the music applet is closed, therefor restore it
+                    self.send_position()
+                else:
+                    print("unknown event")
+
+    def send_metadata(self, metadata=None , /, title="", artist="", album="", total_length=0):
+        if metadata:
+            self.track_chrc.write_value(bytes(metadata['xesam:title'], 'utf-8'))
+            self.artist_chrc.write_value(bytes(metadata['xesam:artist'][0], 'utf-8'))
+            self.album_chrc.write_value(bytes(metadata['xesam:album'], 'utf-8'))
+            self.total_length_chrc.write_value((metadata['mpris:length']//(1000*1000)).to_bytes(4, byteorder='big'))
```

Comment:
```suggestion
            try:
                self.album_chrc.write_value(bytes(metadata['xesam:album'], 'utf-8'))
            except KeyError as error:
                print("No album art found at xesam:album")
            try:
                self.total_length_chrc.write_value((metadata['mpris:length']//(1000*1000)).to_bytes(4, byteorder='big'))
            except KeyError as error:
                print("No track length at mpris:length")

```

I wasn't finding this [in Shortwave](https://gitlab.gnome.org/World/Shortwave) when trying SomaFM radio stations. Another option is to use `metadata.get()` and provide a default value, but I'm not sure what that would be.

Reply:
Thanks for you suggestions,
I was testing this with Lollypop, GnomeMusic and vlc in which the album field always contained a default. I implemented your suggestions and added default values. I also tested Shortwave, which worked fine for me. I'm too on InfiniTime 1.3.0

Reply:
Awesome, I’ll try those out too 💪

(Probably something on my end)